compiler/damlc/tests/daml-test-files/ChoiceArgField.EXPECTED.desugared-daml

```diff
@@ -0,0 +1,140 @@
+    where
+        _ = cid
+instance DA.Internal.Desugar.HasCreate Asset where
+  create = GHC.Types.primitive @"UCreate"
+instance DA.Internal.Desugar.HasFetch Asset where
+  fetch = GHC.Types.primitive @"UFetch"
+instance DA.Internal.Desugar.HasToAnyTemplate Asset where
+  _toAnyTemplate = GHC.Types.primitive @"EToAnyTemplate"
+instance DA.Internal.Desugar.HasFromAnyTemplate Asset where
+  _fromAnyTemplate = GHC.Types.primitive @"EFromAnyTemplate"
+instance DA.Internal.Desugar.HasTemplateTypeRep Asset where
+  _templateTypeRep = GHC.Types.primitive @"ETemplateTypeRep"
+instance DA.Internal.Desugar.HasIsInterfaceType Asset where
+  _isInterfaceType _ = DA.Internal.Desugar.False
+instance DA.Internal.Desugar.HasExercise Asset DA.Internal.Desugar.Archive (()) where
+  exercise = GHC.Types.primitive @"UExercise"
+instance DA.Internal.Desugar.HasToAnyChoice Asset DA.Internal.Desugar.Archive (()) where
+  _toAnyChoice = GHC.Types.primitive @"EToAnyChoice"
+instance DA.Internal.Desugar.HasFromAnyChoice Asset DA.Internal.Desugar.Archive (()) where
+  _fromAnyChoice = GHC.Types.primitive @"EFromAnyChoice"
+instance DA.Internal.Desugar.HasExercise Asset ChoiceWithArgField (Int) where
+  exercise = GHC.Types.primitive @"UExercise"
+instance DA.Internal.Desugar.HasToAnyChoice Asset ChoiceWithArgField (Int) where
+  _toAnyChoice = GHC.Types.primitive @"EToAnyChoice"
+instance DA.Internal.Desugar.HasFromAnyChoice Asset ChoiceWithArgField (Int) where
+  _fromAnyChoice = GHC.Types.primitive @"EFromAnyChoice"
+instance DA.Internal.Desugar.HasExercise Asset ChoiceWithoutArgField (Int) where
+  exercise = GHC.Types.primitive @"UExercise"
+instance DA.Internal.Desugar.HasToAnyChoice Asset ChoiceWithoutArgField (Int) where
+  _toAnyChoice = GHC.Types.primitive @"EToAnyChoice"
+instance DA.Internal.Desugar.HasFromAnyChoice Asset ChoiceWithoutArgField (Int) where
+  _fromAnyChoice = GHC.Types.primitive @"EFromAnyChoice"
+_choice_AssetArchive :
+  (Asset
+   -> DA.Internal.Desugar.Archive -> [DA.Internal.Desugar.Party],
+   DA.Internal.Desugar.ContractId Asset
+   -> Asset
+      -> DA.Internal.Desugar.Archive -> DA.Internal.Desugar.Update (()),
+   DA.Internal.Desugar.Consuming Asset,
+   DA.Internal.Desugar.Optional (Asset
+                                 -> DA.Internal.Desugar.Archive -> [DA.Internal.Desugar.Party]))
+_choice_AssetArchive
+  = (\ this _ -> DA.Internal.Desugar.signatory this, 
+     \ _ _ _ -> pure (), DA.Internal.Desugar.Consuming, 
+     DA.Internal.Desugar.None)
+_choice_AssetChoiceWithArgField :
+  (Asset -> ChoiceWithArgField -> [DA.Internal.Desugar.Party],
+   DA.Internal.Desugar.ContractId Asset
+   -> Asset -> ChoiceWithArgField -> DA.Internal.Desugar.Update (Int),
+   DA.Internal.Desugar.Consuming Asset,
+   DA.Internal.Desugar.Optional (Asset
+                                 -> ChoiceWithArgField -> [DA.Internal.Desugar.Party]))
+_choice_AssetChoiceWithArgField
+  = (\ this@Asset {..} ChoiceWithArgField {..}
```

Comment:
Here we see the effect of this PR. The second argument is _not_ bound to `arg` since that would clash with the choice record's field of the same name